packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -393,13 +394,17 @@ module.exports = function(
 
     if (instance.state !== oldState) {
       if (__DEV__) {
-        warning(
-          false,
-          '%s.componentWillReceiveProps(): Assigning directly to ' +
-            "this.state is deprecated (except inside a component's " +
-            'constructor). Use setState instead.',
-          getComponentName(workInProgress),
-        );
+        const componentName = getComponentName(workInProgress) || 'Component';
```

Comment:
We should put this assignment inside the `if` block. So that we don't spend time doing it if we don't use the result.

Reply:
but the if condition is based on this value. So, we need this statement here

Reply:
Oops, you're right. I didn't realize this is only called when we *already* know we're gonna warn. Disregard my previous comment.